### PR TITLE
Add minimize feature for ensemble selector

### PR DESCRIPTION
**Pull request by autt**
Towards #70

### CODE REVIEW

File: ertviz/controllers/ensemble_selector_controller.py [hunk 1]
```diff
@@ -46,24 +46,17 @@ def _construct_edge(ensemble_src, ensemble_target):
...
-            Input("url", "pathname"),
             Input(parent.uuid("ensemble-selection-store"), "data"),
+            Input(parent.uuid("ensemble-selector-button"), "n_clicks"),
         ],
         [
             State(parent.uuid("ensemble-selector"), "elements"),
         ],
     )
-    def update_ensemble_selector(query, selected_ensembles, elements):
+    def update_ensemble_selector(selected_ensembles, n_clicks, elements):
```
Comment (xjules) [inline ertviz/controllers/ensemble_selector_controller.py:59]: What's the `n_clicks` used for?

File: ertviz/controllers/ensemble_selector_controller.py [hunk 2]
```diff
@@ -100,3 +93,40 @@ def update_ensemble_selection(selected_nodes):
...
         return data
+
+    @app.callback(
+        [
+            Output(parent.uuid("ensemble-selector"), "className"),
+            Output(parent.uuid("ensemble-selector-container"), "className"),
+            Output(parent.uuid("ensemble-selector-button"), "children"),
+        ],
+        [
+            Input(parent.uuid("ensemble-selector-button"), "n_clicks"),
```
Comment (xjules) [inline ertviz/controllers/ensemble_selector_controller.py:104]: Could this be replaced by a checkbox instead? You wouldn't need to count the number of clicks

File: ertviz/controllers/ensemble_selector_controller.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ManInFez) [reply ertviz/controllers/ensemble_selector_controller.py:59]: just to update layout when changing view

[the remark this answers is not in the document]
Comment (autt) [reply ertviz/controllers/ensemble_selector_controller.py:59]: We want this function to be triggered when n_clicks is updated (button is clicked), and hence n_clicks is an input even though it is not used. Maybe your suggestion of replacing the button with a checkbox would be a better solution :smile:

File: ertviz/controllers/ensemble_selector_controller.py (hunk 2 above)
Comment (ManInFez) [reply ertviz/controllers/ensemble_selector_controller.py:104]: The whole view changes place and size and label. I think that checkboxes makes more sense when it is standing still. n_clicks is the default click trigger for buttons.

Comment (xjules) [reply ertviz/controllers/ensemble_selector_controller.py:101]: Ok, you made your case :)

File: ertviz/controllers/ensemble_selector_controller.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (xjules) [reply ertviz/controllers/ensemble_selector_controller.py:56]: @ManInFez might be right as checkbox makes wrong impression when changing place. Still need to test it on my machine.

[the remark this answers is not in the document]
Comment (autt) [reply ertviz/controllers/ensemble_selector_controller.py:56]: Yes, I also prefer the button with changing label over a checkbox

**Comment by xjules** [comment]
Is it intended not see selected ensembles in minimize version?

**Comment by autt** [comment]
> Is it intended not see selected ensembles in minimized version?

Not really, but I am experiencing some issues with the coloring of selected ensembles both in the expanded and minimized version

**Comment by xjules** [comment]
After doing maximization again I lost colors for selected ensembles completely. Also when selecting from anew just default color is active:
image

**Review by oysteoh — APPROVED** [review]
Seems to work fine! I think modifications and improvements can be added later ( context to the data and such)

**Pull request merged**